### PR TITLE
generate url decode benchmark input in device

### PR DESCRIPTION
**Pull request by karthikeyann**
Use device functions to move input generation to device in url_decode benchmark.

Splitting PR https://github.com/rapidsai/cudf/pull/10109 for review

### CODE REVIEW

File: cpp/benchmarks/string/url_decode.cu [hunk 1]
```diff
@@ -16,55 +16,68 @@
...
+    engine.discard(begin);
+    for (auto i = begin; i < end; ++i) {
+      if (esc_seq_dist(engine) < esc_seq_chance and i < end - 3) {
+        chars[i]     = '%';
+        chars[i + 1] = '2';
+        chars[i + 2] = '0';
+        i += 2;
       } else {
-        str.push_back('a');
+        chars[i] = 'a';  // + (i % 26);
```
Comment (davidwendt) [inline cpp/benchmarks/string/url_decode.cu:59]:
```suggestion
        chars[i] = 'a';
```

Comment (karthikeyann) [reply cpp/benchmarks/string/url_decode.cu:59]: Kept it here as a hint for suggestion during review. will remove this.

**Review by davidwendt — APPROVED** [review]

File: cpp/benchmarks/string/url_decode.cu (hunk 1 above)
Comment (hyperbolic2346) [reply cpp/benchmarks/string/url_decode.cu:59]: Are you suggesting that the line should be `chars[i] = 'a' + (i % 26)` or is the desired output "aaaaa%20aaaa"?

Comment (karthikeyann) [reply cpp/benchmarks/string/url_decode.cu:59]: AFAIK, 'aaaa' is just arbitrary string. So having `chars[i] = 'a' + (i % 26)` would make the string slightly non-homogenous.
In this case, it won't any difference in benchmark results.

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by karthikeyann** [comment]
@gpucibot merge

**Pull request merged**